### PR TITLE
flex data source protocol refactor

### PR DESCRIPTION
**Pull request by schrockblock**

### CODE REVIEW

File: FlexTitledDataSource.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//
+//  FlexTitledDataSource.swift
+//  FlexDataSource
+//
+//  Created by Calvin Collins on 4/19/21.
+//
+
+import Foundation
```
Comment (schrockblock) [inline FlexTitledDataSource.swift:8]: Let's delete these files

File: FlexHeaderDataSource.swift [hunk 2]
```diff
@@ -0,0 +1,8 @@
+//
+//  FlexHeaderDataSource.swift
+//  FlexDataSource
+//
+//  Created by Calvin Collins on 4/19/21.
+//
+
+import Foundation
```
Comment (schrockblock) [inline FlexHeaderDataSource.swift:8]: Ditto

File: FlexDataSource/Classes/FlexUntitledDataSource.swift [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+//  FlexDataSource.swift
+//  FlexDataSource
+//
+//  Created by Elliot Schrock on 2/10/18.
+//
+
+import UIKit
+import fuikit
+
+open class FlexUntitledDataSource: FUITableViewDataSource, FlexDataSource {
```
Comment (schrockblock) [inline FlexDataSource/Classes/FlexUntitledDataSource.swift:11]: Did we want to change this to Simple?

File: FlexDataSource/Classes/FlexUntitledDataSource.swift [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        }
+        return UITableViewCell()
+    }
+}
+
+public let deselectRow: (UITableView, IndexPath) -> Void = { $0.deselectRow(at: $1, animated: true) }
+
+// MARK: - Tapping
+
+public extension FlexUntitledDataSource {
```
Comment (schrockblock) [inline FlexDataSource/Classes/FlexUntitledDataSource.swift:88]: Do these work at the protocol level? That is, do we need to extend FlexUntitledDataSource or could we extend the protocol?

File: FlexDataSource/Classes/FlexUntitledDataSource.swift [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            if let item = self.sections?[indexPath.section].items?[indexPath.row] {
+                onTap(item)
+            }
+        }
+    }
+}
+
+// MARK: - Swiping
+
+public extension FlexUntitledDataSource {
```
Comment (schrockblock) [inline FlexDataSource/Classes/FlexUntitledDataSource.swift:108]: Ditto on extending

File: FlexDataSource/Classes/Array+Flex.swift [hunk 6]
```diff
@@ -8,8 +8,8 @@
 import Foundation
 
 public extension Array where Element: FlexDataSourceItem {
-    func flexDataSource() -> FlexDataSource {
-        let ds = FlexDataSource()
+    func flexDataSource() -> FlexUntitledDataSource {
```
Comment (schrockblock) [inline FlexDataSource/Classes/Array+Flex.swift:11]: Could this return the protocol? Do the consumers need to know it's the base version?

File: FlexDataSource.swift [hunk 7]
```diff
@@ -0,0 +1,8 @@
+//
+//  FlexDataSource.swift
+//  FlexDataSource
+//
+//  Created by Calvin Collins on 4/19/21.
+//
+
+import Foundation
```
Comment (schrockblock) [inline FlexDataSource.swift:8]: Delete file

**Review by schrockblock — COMMENTED** [review]
While we're in here, @cjc8 could you make this work with SPM?

File: FlexDataSource/Classes/FlexUntitledDataSource.swift (hunk 4 above)
Comment (cjc8) [reply FlexDataSource/Classes/FlexUntitledDataSource.swift:88]: Ah, good idea i'll try that.

**Review by schrockblock — COMMENTED** [review]
Couple small things, then it's good to go!

**Pull request merged**